ui/src/components/credentials/CredentialLinkQR.tsx

```diff
@@ -0,0 +1,310 @@
+  const { lg } = Grid.useBreakpoint();
+  const { linkID } = useParams();
+
+  const createCredentialQR = useCallback(
+    async (signal: AbortSignal) => {
+      if (linkID) {
+        setShareCredentialQRCode({ status: "loading" });
+
+        const response = await createCredentialLinkQRCode({ env, linkID, signal });
+
+        if (response.isSuccessful) {
+          setShareCredentialQRCode({ data: response.data, status: "successful" });
+        } else {
+          if (!isAbortedError(response.error)) {
+            setShareCredentialQRCode({ error: response.error, status: "failed" });
+          }
+        }
+      }
+    },
+    [linkID, env]
+  );
+
+  useEffect(() => {
+    const { aborter } = makeRequestAbortable(createCredentialQR);
+
+    return aborter;
+  }, [createCredentialQR]);
+
+  useEffect(() => {
+    const checkCredentialQRCode = async () => {
+      if (isAsyncTaskDataAvailable(shareCredentialQRCode) && linkID) {
+        const response = await getCredentialLinkQRCode({
+          env,
+          linkID,
+          sessionID: shareCredentialQRCode.data.sessionID,
+        });
+
+        if (response.isSuccessful) {
+          if (response.data.status !== "pending") {
+            setCredentialQRCheck({ data: response.data, status: "successful" });
+
+            const { proofTypes } = shareCredentialQRCode.data.linkDetail;
+
+            if (proofTypes.includes("SparseMerkleTreeProof")) {
+              void message.info("Issuance process started");
+            }
+
+            if (proofTypes.includes("BJJSignature2021")) {
+              void message.success("Credential sent");
+            }
+          }
+        } else {
+          setCredentialQRCheck({ error: response.error, status: "failed" });
+
+          void message.error(response.error.message);
+        }
+      }
+    };
+
+    const checkQRCredentialStatusTimer = setInterval(() => {
+      if (
+        (isAsyncTaskDataAvailable(credentialQRCheck) &&
+          credentialQRCheck.data.status !== "pending") ||
+        hasAsyncTaskFailed(credentialQRCheck)
+      ) {
+        clearInterval(checkQRCredentialStatusTimer);
+      } else {
+        void checkCredentialQRCode();
+      }
+    }, POLLING_INTERVAL);
+
+    return () => clearInterval(checkQRCredentialStatusTimer);
+  }, [shareCredentialQRCode, linkID, credentialQRCheck, env]);
+
+  const onStartAgain = () => {
+    makeRequestAbortable(createCredentialQR);
+    setCredentialQRCheck({ status: "pending" });
+  };
+
+  const hasFailed = hasAsyncTaskFailed(shareCredentialQRCode)
+    ? shareCredentialQRCode.error
+    : hasAsyncTaskFailed(credentialQRCheck)
+    ? credentialQRCheck.error
+    : undefined;
+
+  if (hasFailed) {
+    if (hasFailed.status === 404) {
+      return (
+        <Space align="center" direction="vertical" size="large">
+          <Avatar className="avatar-color-error" icon={<QRIcon />} size={56} />
+
+          <Typography.Title level={2}>Claim link is invalid</Typography.Title>
```

Comment:
This is ["credential"](https://www.figma.com/file/BE3tUbvtSB2EqmJsybpeFC/Polygon-ID-Platform?node-id=7412-174560&t=9LVpltG1CrqTkdxR-0).